errors.go

```diff
@@ -54,23 +58,48 @@ func (e ErrHederaNetwork) Error() string {
 	return fmt.Sprintf("transport error occurred while accessing the Hedera network: %s", e.Error())
 }
 
-type ErrHederaStatus struct {
```

Comment:
I thought we're not deleting the old type. Instead I thought the new error types would extend this error type. IIRC, it would be something like this. This should allow you access to the `Status` underneath.
```go
type ErrHederaPrecheckStatus struct {
    ErrHederaStatus
}
```

Reply:
This can't be done properly in Go and also isn't needed since Go isn't on 1.0.0 yet.